CONTRIBUTING.md

```diff
@@ -1,15 +1,143 @@
+- [#prow]: for discussions related to the use or development of [`prow`](/prow)
+
+## Workflow
+
+In general we use the same pull request (PR) and review workflow used by the
+rest of the Kubernetes project. This means the [use of a CLA][cla], [code
+review by reviewers and approvers listed in OWNERS files][owners], and tests
+that automatically exercise code or enforce conventions. Many PRs will result
+in changes automatically being deployed.
+
+Some conventions that are specific to this repo:
+
+- For large code changes, please write a design doc or [KEP] and get signoff
+  from SIG Testing before trying to land code. If you're not sure what a KEP
+  should look like, [kuberenetes/enhancements/keps/sig-testing] has some
+  examples. If you're not sure what "large" means, [come ask us](#contact)
+- We find it polite to apply the `do-not-merge/hold` label via the `/hold`
+  command when in doubt about whether/when a PR should merge. If we don't
+  explain why we're adding a `/hold`, this is usually because the PR will
+  cause changes to be deployed on merge, and we want the PR author to
+  `/hold cancel` when they are available to monitor the results of their
+  change going live.
+- Many of us use [gubernator.k8s.io/pr] to keep track of which PRs require
+  our attention. Use of the `/cc @person` or `/assign @person` commands is the
+  most effective way to put PRs on our radar. You can also
+  [contact us](#contact) on slack.
+
+## Issue Triage
+
+Issues are ideally labeled with:
+
+- milestone: during which release cycle do we plan on working on this issue
+- `sig/foo`: which SIG owns this work
+- `area/foo`: which subproject or code is this issue related to
+- `kind/foo`: which kind of work is this issue describing
+- `priority/foo`: how important is this issue
+
+For example, an issue related to cleaning up and consolidating (`kind/cleanup`)
+release-related (`sig/release`) jobs and dashboards (`area/config`) for the
+v1.16 cycle (`milestone: v1.16`) that may not get completed by the end of the
+cycle if more important or more urgent arises (`priority/important-longterm`).
+
+We try to have a non-stale pool of issues that are available for new
+contributors who want to help out but aren't sure what to work on or where to
+get started:
+
+- [`label:"good first issue"`][good-first-issue] - triaged per [Good First Issue docs][good-first-issue-docs]
+- [`label:"help wanted"`][help-wanted] - triaged per [Help Wanted docs][help-wanted-docs]
+
+## Guides
+
+### Build and Test
+
+We use [`bazel`][bazel] to build, test and deploy code in this repo. In most
+cases you can get by just fine with:
+
+```
+bazel build //...
+bazel test //...
+```
+
+If you modify Go code, run `./hack/update-bazel.sh` to keep `BUILD.bazel` files
+up to date
+
+### Dependency Management
+
+Please see [/docs/deps.md]
+
+### Prow Jobs
+
+The majority of the Kubernetes project is tested via prowjobs that run on
```

Comment:
It would be helpful to breifly introduce that Prow is the CI thing that we use and has jobs before throwing out the term prowjobs. As a new contributor I have no idea what this sentence means and even if I go to config/jobs/README.md I still won't know what Prow is without clicking through more links.

Reply:
I see we explain these pretty well in the root README, but the contributing doc is also surfaced in other places, we should probably at least place a callback to the root README somewhere early in this doc.

Reply:
Link to root README before these specific headings